clients/venice-producer/src/main/java/com/linkedin/venice/producer/VeniceProducerMetrics.java

```diff
@@ -0,0 +1,92 @@
+package com.linkedin.venice.producer;
+
+import com.linkedin.venice.stats.AbstractVeniceStats;
+import com.linkedin.venice.stats.Gauge;
+import com.linkedin.venice.stats.TehutiUtils;
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.Sensor;
+import io.tehuti.metrics.stats.Avg;
+import io.tehuti.metrics.stats.Max;
+import io.tehuti.metrics.stats.OccurrenceRate;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+public class VeniceProducerMetrics extends AbstractVeniceStats {
+  private final boolean enableMetrics;
+  private Sensor operationSensor = null;
+  private Sensor putOperationSensor = null;
+  private Sensor deleteOperationSensor = null;
+  private Sensor updateOperationSensor = null;
+  private Sensor successOperationSensor = null;
+  private Sensor failedOperationSensor = null;
+  private Sensor produceLatencySensor = null;
+  private Sensor pendingOperationSensor = null;
+
+  private final AtomicInteger pendingOperationCounter = new AtomicInteger(0);
+
+  public VeniceProducerMetrics(MetricsRepository metricsRepository, String storeName) {
+    super(metricsRepository, storeName);
+
+    if (metricsRepository != null) {
+      enableMetrics = true;
+      operationSensor = registerSensor("write_operation", new OccurrenceRate());
+      putOperationSensor = registerSensor("put_operation", new OccurrenceRate());
+      deleteOperationSensor = registerSensor("delete_operation", new OccurrenceRate());
+      updateOperationSensor = registerSensor("update_operation", new OccurrenceRate());
+
+      successOperationSensor = registerSensor("success_write_operation", new OccurrenceRate());
+      failedOperationSensor = registerSensor("failed_write_operation", new OccurrenceRate());
+      String produceLatencySensorName = "produce_to_durable_buffer_latency";
+      produceLatencySensor = registerSensor(
+          produceLatencySensorName,
+          TehutiUtils.getPercentileStat(getName() + AbstractVeniceStats.DELIMITER + produceLatencySensorName));
+
+      pendingOperationSensor = registerSensor("pending_write_operation", new Avg(), new Max(), new Gauge());
```

Comment:
Do we need the `Gauge`? I don't think it's intended to be used in this way...

Reply:
We probably don't need it. I added it to test the metrics code. With just `Avg` and `Max`, the test cases won't make much sense.

I can do two things:
1. Remove that Gauge metric and the test lines for it
2. Add the Gauge only for tests. I'm not keen on doing this as it doesn't seem to add any value

Reply:
The test cases I've written are like:
```
Assert.assertEquals(metricsRepository.getMetric(PENDING_OPERATION_METRIC_NAME).value(), 0.0);
```

These use the `Gauge` currently. The value for the `Avg` is 0.5 and for the `Max` it is 1. `Gauge` will give us the last measured metric value (obv. assuming that the test case doesn't run for too long that the metric value gets reset)

Reply:
Replaced `Gauge` and `Avg` with `Min` since `Avg` is slightly tricky to reason about